Failure to deserialize Scala 3 enum case objects

Reproducer (Scala 3 only):
```scala
import io.altoo.serialization.kryo.scala.ScalaKryoSerializer
import com.typesafe.config.{ Config, ConfigFactory }

enum A {
  case B()
  case C()
  case D
}

val config = ConfigFactory.defaultReference()
val ser = new ScalaKryoSerializer(config, getClass.getClassLoader)
ser.deserialize[A](ser.serialize(A.D).get).get
```
Also on Scastie: https://scastie.scala-lang.org/063ni8hLTI20kpV9U7npdg

It works fine with B and C, but getting a runtime exception with D.

```
Caused by: java.lang.NoSuchMethodException: Playground$A.valueOf(java.lang.String)
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2675)
	at io.altoo.serialization.kryo.scala.serializer.ScalaEnumNameSerializer.read(ScalaEnumNameSerializer.scala:15)
	at io.altoo.serialization.kryo.scala.serializer.ScalaEnumNameSerializer.read(ScalaEnumNameSerializer.scala:11)
	at com.esotericsoftware.kryo.Kryo.readClassAndObject(Kryo.java:877)
	at io.altoo.serialization.kryo.scala.KryoSerializerBackend.fromBinary(KryoSerializerBackend.scala:61)
	at io.altoo.serialization.kryo.scala.KryoSerializer.fromBinaryInternal(KryoSerializer.scala:183)
	at io.altoo.serialization.kryo.scala.ScalaKryoSerializer.deserialize$$anonfun$1(ScalaKryoSerializer.scala:23)
	at scala.util.Try$.apply(Try.scala:210)
	at io.altoo.serialization.kryo.scala.ScalaKryoSerializer.deserialize(ScalaKryoSerializer.scala:23)
```

--- a/core/src/main/scala-3/io/altoo/serialization/kryo/scala/serializer/ScalaEnumNameSerializer.scala
+++ b/core/src/main/scala-3/io/altoo/serialization/kryo/scala/serializer/ScalaEnumNameSerializer.scala
@@ -11,8 +11,16 @@ class ScalaEnumNameSerializer[T <: EnumValue] extends Serializer[T] {
   def read(kryo: Kryo, input: Input, typ: Class[? <: T]): T = {
     val clazz = kryo.readClass(input).getType
     val name = input.readString()
-    // using value instead of ordinal to make serialization more stable, e.g. allowing reordering without breaking compatibility
-    clazz.getDeclaredMethod("valueOf", classOf[String]).invoke(null, name).asInstanceOf[T]
+
+    try {
+      // using value instead of ordinal to make serialization more stable, e.g. allowing reordering without breaking compatibility
+      clazz.getDeclaredMethod("valueOf", classOf[String]).invoke(null, name).asInstanceOf[T]
+    } catch {
+      case _: java.lang.NoSuchMethodException =>
+        // work around Scala 3 ADT-like enums missing valueOf method
+        val objectClazz = Class.forName(clazz.getName + "$")
+        objectClazz.getDeclaredField(name).get(null).asInstanceOf[T]
+    }
   }
 
   def write(kryo: Kryo, output: Output, obj: T): Unit = {

--- a/core/src/test/scala-3/io/altoo/serialization/kryo/scala/serializer/ScalaEnumSerializationTest.scala
+++ b/core/src/test/scala-3/io/altoo/serialization/kryo/scala/serializer/ScalaEnumSerializationTest.scala
@@ -1,9 +1,7 @@
 package io.altoo.serialization.kryo.scala.serializer
 
 import com.esotericsoftware.kryo.util.{DefaultClassResolver, ListReferenceResolver}
-import io.altoo.serialization.kryo.scala.ScalaVersionSerializers
-import io.altoo.serialization.kryo.scala.serializer.{ScalaEnumNameSerializer, ScalaKryo}
-import io.altoo.serialization.kryo.scala.testkit.{AbstractKryoTest, KryoSerializationTesting}
+import io.altoo.serialization.kryo.scala.testkit.KryoSerializationTesting
 import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 
@@ -17,34 +15,52 @@ object ScalaEnumSerializationTest {
   case class EmbeddedEnum(sample: Sample) {
     def this() = this(null)
   }
+
+  enum SimpleADT {
+    case A()
+    case B
+  }
 }
 
-class ScalaEnumSerializationTest  extends AnyFlatSpec with Matchers with KryoSerializationTesting {
+class ScalaEnumSerializationTest extends AnyFlatSpec with Matchers with KryoSerializationTesting {
   import ScalaEnumSerializationTest.*
 
   val kryo = new ScalaKryo(new DefaultClassResolver(), new ListReferenceResolver())
   kryo.setRegistrationRequired(false)
   kryo.addDefaultSerializer(classOf[scala.runtime.EnumValue], new ScalaEnumNameSerializer[scala.runtime.EnumValue])
 
-
   behavior of "Kryo serialization"
 
-  it should "reoundtrip enum" in {
+  it should "round trip enum" in {
     kryo.setRegistrationRequired(false)
 
     testSerializationOf(Sample.B)
   }
 
-  it should "reoundtrip external enum" in {
+  it should "round trip external enum" in {
     kryo.setRegistrationRequired(false)
 
     testSerializationOf(io.altoo.external.ExternalEnum.A)
   }
 
-  it should "reoundtrip embedded enum" in {
+  it should "round trip embedded enum" in {
     kryo.setRegistrationRequired(false)
     kryo.register(classOf[EmbeddedEnum], 46)
 
     testSerializationOf(EmbeddedEnum(Sample.C))
+  }
+
+  it should "round trip adt enum class using generic field serializer" in {
+    kryo.setRegistrationRequired(false)
+    kryo.register(classOf[SimpleADT], 47)
+
+    testSerializationOf(SimpleADT.A)
+  }
+
+  it should "round trip adt enum object using enum serializer" in {
+    kryo.setRegistrationRequired(false)
+    kryo.register(classOf[SimpleADT], 47)
+
+    testSerializationOf(SimpleADT.B)
   }
 }

